packages/nodes-base/nodes/ActionNetwork/ActionNetwork.node.ts

```diff
@@ -0,0 +1,525 @@
+				name: 'comments',
+				type: 'string',
+				required: false,
+				default: '',
+				displayOptions: { show: { related_resource: ['signatures'], method: ['POST'] } }
+			},
+			{
+				displayName: 'Send Autorespond Email',
+				name: 'autorespond',
+				type: 'boolean',
+				required: false,
+				default: true,
+				displayOptions: {
+					show: {
+						related_resource: ['attendances', 'submissions', 'signatures'],
+						method: ['POST']
+					}
+				}
+			},
+			{
+				displayName: 'Donation Recipients',
+				name: 'recipients',
+				type: 'fixedCollection',
+				default: '',
+				placeholder: 'Add Donation Recipient',
+				displayOptions: { show: { related_resource: ['donations'], method: ['POST'] } },
+				typeOptions: {
+					multipleValues: true,
+				},
+				options: [
+					{
+						name: 'recipients',
+						displayName: 'Recipients',
+						values: [
+							{
+								displayName: 'Name',
+								name: 'display_name',
+								type: 'string',
+								default: '',
+							},
+							{
+								displayName: 'Amount',
+								name: 'amount',
+								type: 'number',
+								default: 0,
+							},
+						]
+					}
+				]
+			},
+			{
+				displayName: 'Outreach Targets',
+				name: 'targets',
+				type: 'fixedCollection',
+				default: '',
+				placeholder: 'Add Donation Recipient',
+				displayOptions: { show: { related_resource: ['outreaches'], method: ['POST'] } },
+				typeOptions: {
+					multipleValues: true,
+				},
+				options: [
+					{
+						name: 'targets',
+						displayName: 'Targets',
+						values: [
+							{
+								displayName: 'First / Given Name',
+								name: 'given_name',
+								type: 'string',
+								default: '',
+							},
+							{
+								displayName: 'Last / Family Name',
+								name: 'family_name',
+								type: 'string',
+								default: '',
+							},
+						]
+					}
+				]
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+		let responseData;
+		const version = this.getNodeParameter('version', 0) as string;
+		const resource = this.getNodeParameter('resource', 0) as string;
+		const method = this.getNodeParameter('method', 0) as string;
+		for (let i = 0; i < items.length; i++) {
+			let url = `/api/${version}/${resource}`
+
+			// Specify a resource
+			const resource_id = this.getNodeParameter('resource_id', i, '') as any
+			if (resource_id) {
+				url += `/${resource_id}`
+			}
+
+			// Specify a related resource
+			const related_resource = this.getNodeParameter('related_resource', i, '') as any
+			if (resource_id) {
+				url += `/${related_resource}`
+			} else if (resource === 'messages' && !!resource_id) {
+				url += '/send'
+			}
+
+			// Specificy a specific related resource
+			const related_resource_id = this.getNodeParameter('related_resource_id', i, '') as any
+			if (resource_id) {
+				url += `/${related_resource_id}`
+			}
+
```

Comment:
The rationale for the `resource` + `related_resource` UI system is that it builds up the URL fairly straightforwardly, without too much logic which might make things messy.